languages/c/little-shop-of-horrors/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.16.5)
+project(lsoh C)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wcast-align")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wwrite-strings")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wunreachable-code")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-strict-aliasing")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-format-nonliteral")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pipe -g -O3 -Werror")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=native")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pedantic -Wall")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-missing-braces -Wextra")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-missing-field-initializers -Wformat=2")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wswitch-default -Wswitch-enum")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wcast-align -Wpointer-arith")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wbad-function-cast -Wstrict-overflow=5")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wstrict-prototypes -Winline")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wundef -Wnested-externs -Wcast-qual")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wshadow -Wunreachable-code")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wlogical-op -Wfloat-equal")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wstrict-aliasing=2")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wredundant-decls")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wold-style-definition -Werror")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ggdb3")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-omit-frame-pointer")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -ffloat-store -fno-common")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstrict-aliasing")
+set(CMAKE_C_FLAGS_DEBUG "-g ${CMAKE_C_FLAGS}")
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS}")
+
+## Dependencies (N/A)
+
+## Sources
+
+add_executable(statx-example src/statx-example.c)
```

Comment:
You might want to read up on modern cmake.

https://cliutils.gitlab.io/modern-cmake/

Reply:
Thanks! I haven't gotten around to really fixing that and this is a one off so I'm not really sure I mind too much :P